pkg/runtime/runtime.go

```diff
@@ -538,13 +538,13 @@ func (a *DaprRuntime) beginComponentsUpdates() error {
 	go func() {
 		stream, err := a.operatorClient.ComponentUpdate(context.Background(), &emptypb.Empty{})
 		if err != nil {
-			log.Errorf("error from operator stream: %s", err)
+			log.Fatalf("error from operator stream: %s", err)
```

Comment:
Is there a way we can signal the runtime to close instead of using Fatal (which causes a panic under the hood)?

Reply:
> Is there a way we can signal the runtime to close instead of using Fatal (which causes a panic under the hood)?

I agree that Fatal is not the best way to quit the runtime. Nevertheless, I had recourse to it because:
 - the use of Fatal is pretty common in the runtime, including in a few other places in runtime.go
 - it is the least disruptive way to address this issue

I would suggest that we create another issue to eliminate all occurrences of Fatal in dapr/dapr

Reply:
In this case, do we want Dapr to crash? If this is a non-recoverable error and Daprd must crash then this is the right thing to do. I don't think we should just remove Fatals from the code - it must be used when it makes sense to use it.

Reply:
I would agree if there were not other go routines running (started by `initRuntime`). IMO I'd like to see the usage of a cancellation context or something like that to cleanly shutdown anything that has initialized. Let @dmitsh said though, that is potentially a significant restructuring of `runtime.go`. I think we can proceed with this PR given that.

Reply:
@artursouza , yes to "this is a non-recoverable error".
@pkedy , I can create a new issue for proper handling this type of failure.